lib/facter/puppetdb_version.rb

```diff
@@ -2,7 +2,25 @@
   confine { Facter::Util::Resolution.which('puppetdb') }
 
   setcode do
-    output = Facter::Core::Execution.execute('puppetdb --version')
-    output.split(':').last.strip
+    command = 'puppetdb --version'
+    splitter = ':'
+    postion = 'last'
+
+    if Facter.value(:os)['family'] == 'Debian'
+      package_maintainer = Facter::Core::Execution.execute('apt-cache show puppetdb | grep "Maintainer:" | head -1')
```

Comment:
Won't this list all *available* versions of the `puppetdb` package (that is the Debian *and* the Puppetlabs ones) regardless of which version is currently installed (if any)?  I feel like taking the first one will not do what we want 100% of the time.

I don't have a Debian 12 system with PuppetDB at hand to check the output ordering, but on Debian 11 where Puppetlabs packages are available `apt-cache show puppetdb` currently list 10 packages from Puppetlabs, and I don't see something that indicate which one is installed…

I feel like `dpkg -s puppetdb` would provide a similar output but *only for the installed* package.  There filtering on the maintainer as you tdo seems to be the best we can do I suppose :shrug:.